aea/decision_maker/scaffold.py

```diff
@@ -0,0 +1,189 @@
+from aea.decision_maker.base import LedgerStateProxy as BaseLedgerStateProxy
+from aea.decision_maker.base import OwnershipState as BaseOwnershipState
+from aea.decision_maker.base import Preferences as BasePreferences
+from aea.decision_maker.messages.base import InternalMessage
+from aea.decision_maker.messages.transaction import TransactionMessage
+
+
+class OwnershipState(BaseOwnershipState):
+    """Represent the ownership state of an agent."""
+
+    def __init__(self):
+        """
+        Instantiate an ownership state object.
+
+        :param decision_maker: the decision maker
+        """
+        raise NotImplementedError
+
+    def set(self, **kwargs,) -> None:
+        """
+        Set values on the ownership state.
+
+        :param amount_by_currency_id: the currency endowment of the agent in this state.
+        :param quantities_by_good_id: the good endowment of the agent in this state.
+        """
+        raise NotImplementedError
+
+    def apply_delta(self, **kwargs,) -> None:
+        """
+        Apply a state update to the ownership state.
+
+        This method is used to apply a raw state update without a transaction.
+
+        :param
+        :return: None
+        """
+        raise NotImplementedError
+
+    def is_affordable_transaction(self, tx_message: TransactionMessage) -> bool:
+        """
+        Check if the transaction is affordable (and consistent).
+
+        E.g. check that the agent state has enough money if it is a buyer or enough holdings if it is a seller.
+        Note, the agent is the sender of the transaction message by design.
+
+        :param tx_message: the transaction message
+        :return: True if the transaction is legal wrt the current state, false otherwise.
+        """
+        raise NotImplementedError
+
+    def apply_transactions(
+        self, transactions: List[TransactionMessage]
+    ) -> "OwnershipState":
+        """
+        Apply a list of transactions to (a copy of) the current state.
+
+        :param transactions: the sequence of transaction messages.
+        :return: the final state.
+        """
+        raise NotImplementedError
+
+    def __copy__(self) -> "OwnershipState":
+        """Copy the object."""
+        raise NotImplementedError
+
+
+class LedgerStateProxy(BaseLedgerStateProxy):
+    """Class to represent a proxy to a ledger state."""
+
+    def __init__(self):
+        """Instantiate a ledger state proxy."""
+        raise NotImplementedError
+
+    @property
+    def is_initialized(self) -> bool:
+        """Get the initialization status."""
+        raise NotImplementedError
+
+    def is_affordable_transaction(self, tx_message: TransactionMessage) -> bool:
+        """
+        Check if the transaction is affordable on the default ledger.
+
+        :param tx_message: the transaction message
+        :return: whether the transaction is affordable on the ledger
+        """
+        raise NotImplementedError
```

Comment:
With moving the ledger apis to connections, we will have to rethink this..it is not currently used so we can just keep it as is